src/reflection.ts

```diff
@@ -813,7 +816,12 @@ export class Parser {
         if (!subTable) {
           throw new Error(`Malformed message missing table at ${field.name()} positon ${idx}`);
         }
-        result.push(deserializer(subTable));
+
+        const typeTable = deserializer[0](subTable);
```

Comment:
I think this use of the deserializer response tuple might be wrong because the subTable is already the "message table" and does not need further processing to be read?